File: Data Structures & Algorithms/two-integer-sum/submission-5.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
// Key: The number, Value: Its index
    var map = new Dictionary<int, int>();

    for (int i = 0; i < nums.Length; i++) {
        int complement = target - nums[i];

        if (map.ContainsKey(complement)) {
            return new int[] { map[complement], i };
        }

        // Add the current number and index to the map
        // (Use TryAdd or check ContainsKey to handle duplicates if necessary)
        map[nums[i]] = i;
    }

    return Array.Empty<int>();
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-5.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty()) return 0;

        int l = 1; // Con trỏ ghi: bắt đầu từ vị trí thứ 2
        
        for (int r = 1; r < nums.size(); r++) {
            // Nếu tìm thấy một phần tử mới khác với phần tử đứng trước nó
            if (nums[r] != nums[r - 1]) {
                nums[l] = nums[r]; // Ghi đè phần tử mới vào vị trí l
                l++; // Tăng con trỏ ghi
            }
        }

        return l; // Trả về số lượng phần tử duy nhất
    }
};